app/models/highlight.rb

```diff
@@ -19,6 +19,14 @@ class Highlight < ApplicationRecord
 
   before_validation :normalize_color, :normalize_source_ids
 
+  scope :by_source_type, ->(source_type) { where(source_type: source_type) }
+  scope :by_color, ->(color) { where(color: color) }
+  scope :by_user, ->(user_id) { where(user_uuid: user_id) }
+  scope :by_order, ->(order_direction) { order(created_at: order_direction.to_sym) }
+  scope :by_source_parent_ids, ->(source_parent_ids) do
```

Comment:
Some way to clarify that just one of the parent IDs needs to match? `by_any_source_parent_id`?

Reply:
personally, i thought this seemed fine.    right now, you can see the trick i do to build the filter dynamically 

```
    filter_by.each do |key, value|
      highlights = highlights.public_send("by_#{key}", value) if value.present?
    end
```

so if we want this to be by_any_source_parent_ids, it would mean changing the swagger binding to property :any_source_parent_id, so that this by_x trick works.

Reply:
ok, it is fine just make sure to make it clear in the description for this field in `GetHighlights` (which requires splitting that definition out from `Highlight` and `NewHighlight`